host/tests/integration_tests/python/runtime/errors.rs

```diff
@@ -21,23 +21,15 @@ def foo(x: int) -> int:
 
     let udf = python_scalar_udf(CODE).await.unwrap();
 
+    // Signature is exact, so it would have been cached;
```

Comment:
So I think you'll need to replicate the checks from 

https://github.com/influxdata/datafusion-udf-wasm/blob/c586f10ba49fff17ad999e34776cf94c42cdd370/guests/python/src/lib.rs#L68-L91

within the host too (for the cached case).